node_agent/na/nodeagent.go

```diff
@@ -92,12 +101,34 @@ func (na *nodeAgentInternal) Start() {
 		glog.Fatalf("Node Agent is not running on the right platform")
 	}
 
-	for {
+	gracePeriodPercentage := *na.config.CertRenewalGracePeriodPercentage
```

Comment:
I would worth to add some comments about this logic to make it more readable, thanks.